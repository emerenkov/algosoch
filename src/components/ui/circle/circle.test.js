import React from 'react';
import renderer from 'react-test-renderer';
import { Circle } from './circle';
import { ElementStates } from "../../../types/element-states";

//use snapshots for correct test, correct view element
describe('test component Circle', () => {
    it('Circle has not symbol', () => {
        const tree = renderer.create(<Circle />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Circle has symbol (Snapshot)', () => {
        const tree = renderer.create(<Circle letter={'test text'} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Circle with head (Snapshot)', () => {
        const tree = renderer.create(<Circle head={'24'} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Circle with react-element in head', () => {
        const tree = renderer.create(<Circle head={<Circle />} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Circle with tail', () => {
        const tree = renderer.create(<Circle tail={'24'} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Circle с react-element in tail', () => {
        const tree = renderer.create(<Circle tail={<Circle />} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Circle has index', () => {
        const tree = renderer.create(<Circle index={24} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Circle has props isSmall === true', () => {
        const tree = renderer.create(<Circle isSmall />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Circle has default position', () => {
        const tree = renderer.create(<Circle state={ElementStates.Default} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Circle has changing position', () => {
        const tree = renderer.create(<Circle state={ElementStates.Changing} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Circle has modified position', () => {
        const tree = renderer.create(<Circle state={ElementStates.Modified} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
