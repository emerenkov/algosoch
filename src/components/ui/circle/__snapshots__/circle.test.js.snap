// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component Circle Circle has changing position 1`] = `
<div
  className="content "
  data-cy-state="changing"
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head div"
  />
  <div
    className="circle   changing"
    data-testid="circle div"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="testCircle"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail div"
  />
</div>
`;

exports[`test component Circle Circle has default position 1`] = `
<div
  className="content "
  data-cy-state="default"
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head div"
  />
  <div
    className="circle   default"
    data-testid="circle div"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="testCircle"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail div"
  />
</div>
`;

exports[`test component Circle Circle has index 1`] = `
<div
  className="content "
  data-cy-state="default"
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head div"
  />
  <div
    className="circle   default"
    data-testid="circle div"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="testCircle"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    24
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-testid="tail div"
  />
</div>
`;

exports[`test component Circle Circle has modified position 1`] = `
<div
  className="content "
  data-cy-state="modified"
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head div"
  />
  <div
    className="circle   modified"
    data-testid="circle div"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="testCircle"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail div"
  />
</div>
`;

exports[`test component Circle Circle has not symbol 1`] = `
<div
  className="content "
  data-cy-state="default"
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head div"
  />
  <div
    className="circle   default"
    data-testid="circle div"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="testCircle"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail div"
  />
</div>
`;

exports[`test component Circle Circle has props isSmall === true 1`] = `
<div
  className="content "
  data-cy-state="default"
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head div"
  />
  <div
    className="circle  small default"
    data-testid="circle div"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="testCircle"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail div"
  />
</div>
`;

exports[`test component Circle Circle has symbol (Snapshot) 1`] = `
<div
  className="content "
  data-cy-state="default"
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head div"
  />
  <div
    className="circle   default"
    data-testid="circle div"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="testCircle"
    >
      test text
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail div"
  />
</div>
`;

exports[`test component Circle Circle with head (Snapshot) 1`] = `
<div
  className="content "
  data-cy-state="default"
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
    data-testid="head div"
  >
    24
  </div>
  <div
    className="circle   default"
    data-testid="circle div"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="testCircle"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail div"
  />
</div>
`;

exports[`test component Circle Circle with react-element in head 1`] = `
<div
  className="content "
  data-cy-state="default"
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head div"
  >
    <div
      className="content "
      data-cy-state="default"
      data-testid="circle"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="head div"
      />
      <div
        className="circle   default"
        data-testid="circle div"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="testCircle"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="tail div"
      />
    </div>
  </div>
  <div
    className="circle   default"
    data-testid="circle div"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="testCircle"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail div"
  />
</div>
`;

exports[`test component Circle Circle with tail 1`] = `
<div
  className="content "
  data-cy-state="default"
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head div"
  />
  <div
    className="circle   default"
    data-testid="circle div"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="testCircle"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-testid="tail div"
  >
    24
  </div>
</div>
`;

exports[`test component Circle Circle —Å react-element in tail 1`] = `
<div
  className="content "
  data-cy-state="default"
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head div"
  />
  <div
    className="circle   default"
    data-testid="circle div"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="testCircle"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail div"
  >
    <div
      className="content "
      data-cy-state="default"
      data-testid="circle"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="head div"
      />
      <div
        className="circle   default"
        data-testid="circle div"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="testCircle"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="tail div"
      />
    </div>
  </div>
</div>
`;
